runtime/lib/common/system_desc.cpp

```diff
@@ -129,6 +136,11 @@ createChipPhysicalCores(const ::tt::tt_metal::Device *device,
         ::tt::target::Dim2d(physical.y, physical.x));
   }
 
+  sort(worker_cores);
```

Comment:
No need to preserve original logical core order (for logical to physical core mapping idea)?

Reply:
We actually explicitly rely on the ordering to be row major in the compiler. Not only that, but this helps canonicalize the system desc, in case for some reason metal doesn't return worker cores in deterministic order, or it changes between versions.

Reply:
I see, so these are all really just a row-major ordered set of physical cores, and compiler doesn't know or care about what logical cores mapped to them?

Reply:
I removed the sorting of worker cores, left the rest to be sorted.